app/models/content_item.rb

```diff
@@ -152,9 +163,17 @@ def includes_auth_bypass_id?(auth_bypass_id)
   end
 
   def register_routes(previous_item: nil)
-    return if self.schema_name.start_with?("placeholder")
-    return if previous_item && previous_item.route_set == self.route_set
-    self.route_set.register!
+    return unless should_register_routes?(previous_item: previous_item)
+
+    3.times do
+      begin
+        route_set.register!
+        break
+      rescue GdsApi::BaseError => e
+        last_exception = e
+      end
+      raise last_exception
```

Comment:
I looked into how we could test this, unfortunately it's currently not possible to configure WebMock to return a different status code when calling again.

I took the code from https://github.com/jordansissel/software-patterns/tree/master/retry-on-failure/ruby, I assume that `raise` does not break out of the block if `last_exception` is `nil`.